src/seedu/addressbook/data/person/Address.java

```diff
@@ -30,10 +29,17 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
     }
 
     /**
-     * Returns true if a given string is a valid person email.
+     * Returns true if a given string is a valid person address.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    	String[] address = test.split(", "); 
```

Comment:
When you are creating components for a class, you must maintain it as a class variable. When needed it should be accessible after initiation. 

You are using the new classes Block, Street, etc only for validation. This is a very limited usage of the new classes. 

Consider creating each component in the Address.java constructor and assigning the specific values to it there (including validation is a good idea there). This way in a given Address instance the Block, Stree, PostalCode, etc components are there with the specific values. Right now this is not happening..